File: Java/Calendar.java
import java.util.Calendar;
import java.util.GregorianCalendar;

public class CalendarGenerator {
    public static void main(String[] args) {
        // Specifies the year for the calendar
        int year = 2023;

        // Create a GregorianCalendar object
        Calendar cal = new GregorianCalendar();

        // Determines the January 1st of the given year
        cal.set(Calendar.YEAR, year);
        cal.set(Calendar.MONTH, Calendar.JANUARY);
        cal.set(Calendar.DAY_OF_MONTH, 1);

        // Display month name and date in calendar format
        String[] months = {"January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
        int month = 0;
        while (cal.get(Calendar.YEAR) == year) {
            month = cal.get(Calendar.MONTH);
            System.out.println("\n" + months[month] + " " + year);
            System.out.println("Su Mo Tu We Th Fr Sa");

            // Specifies the date for each day of the month
            int dayOfWeek = cal.get(Calendar.DAY_OF_WEEK);
            for (int i = 1; i < dayOfWeek; i++) {
                System.out.print("   ");
            }
            for (int i = 1; i <= cal.getActualMaximum(Calendar.DAY_OF_MONTH); i++) {
                System.out.printf("%2d ", i);
                if (((i + dayOfWeek - 1) % 7 == 0) || (i == cal.getActualMaximum(Calendar.DAY_OF_MONTH))) {
                    System.out.println();
                }
            }

            // Determine the next month
            cal.add(Calendar.MONTH, 1);
        }
    }
}

File: Python/calendar.py
import calendar

def generate_calendar(year):
    # Creates a calendar for a given year
    cal = calendar.Calendar()
    cal.setfirstweekday(calendar.SUNDAY)
    months = cal.yeardatescalendar(year, width=3)

    # Display month name and date in calendar format
    for year_month in months:
        for month_weeks in year_month:
            month_name = calendar.month_name[month_weeks[0].month]
            print(f"\n{month_name} {year}")
            print("Su Mo Tu We Th Fr Sa")
            for week in month_weeks:
                for day in week:
                    if day.month == month_weeks[0].month:
                        print(f"{day.day:2d}", end=" ")
                    else:
                        print("  ", end=" ")
                print()

if __name__ == "__main__":
    generate_calendar(2023)
